src/hooks/useClickOutside.jsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from "react";
+
+const useClickOutside = (ref, handler) => {
+  useEffect(() => {
+    const listener = (event) => {
+      event.stopPropagation();
+      if (!ref.current || ref.current.contains(event.target)) {
+        return;
+      }
+      handler(event);
+    };
+    document.addEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
+    };
+  }, [ref, handler]);
+};
+
+export { useClickOutside };
```

Comment:
This is something new

Reply:
yes , I have just created a hook for modal will close on clicked outside .